src/Analysis/Ast/Impl/Modules/PythonModule.cs

```diff
@@ -164,8 +164,17 @@ internal PythonModule(ModuleCreationOptions creationOptions, IServiceContainer s
                     if (valueType is PythonModule) {
                         return false; // Do not re-export modules.
                     }
-                    // Do not re-export types from typing
-                    return !(valueType?.DeclaringModule is TypingModule) || this is TypingModule;
+                    if (this is TypingModule) {
+                        return true; // Let typing module behave normally.
+                    }
+                    // Do not re-export types from typing. However, do export variables
```

Comment:
Fixes the issue when we did not export variables that were not types from `typing` but rather assignments like `x = Union[int, str]`, thinking that `x` is a type from `typing`.